### PR TITLE
NumPy course improvements

### PR DESCRIPTION
**Pull request by dkillick**
Some improvements to the NumPy course. Includes:

 * Changing the `x` values used in the final exercise (see also #26).
 * Changing the values used in &sect;Slices so that the values are not the same as the indexes.

### CODE REVIEW

**Comment by dkillick** [comment]
@pelson @lbdreyer care to take a look?

File: course_content/notebooks/numpy_intro.ipynb [hunk 1]
```diff
@@ -995,7 +995,7 @@
      "cell_type": "code",
      "collapsed": false,
      "input": [
-      "arr = np.arange(8).reshape(2, 4)\n",
+      "arr = np.arange(2, 18, 2).reshape(2, 4)\n",
```
Comment (pelson) [inline course_content/notebooks/numpy_intro.ipynb]: How about using linspace. I'm not fond of making an array of an expected length without specifying the length itself.

File: course_content/notebooks/numpy_intro.ipynb [hunk 2]
```diff
@@ -1987,7 +1992,7 @@
       "\n",
       "#### Part 1\n",
       "\n",
-      "Create two arrays $x$ and $y$, where $x$ is a linearly spaced array in the interval $[0, 3]$ of length 10, and $y$ represents the function $f(x) = x^2$ sampled at $x$."
+      "Create two arrays $x$ and $y$, where $x$ is a linearly spaced array in the interval $[0, 3]$ of length 11 (for the sake of nice numbers), and $y$ represents the function $f(x) = x^2$ sampled at $x$."
```
Comment (pelson) [inline course_content/notebooks/numpy_intro.ipynb]: The brackets aren't necessary IMO.

**Comment by pelson** [comment]
Glad you've taken this on. Thanks @dkillick!

File: course_content/notebooks/numpy_intro.ipynb (hunk 1 above)
Comment (dkillick) [inline course_content/notebooks/numpy_intro.ipynb]: Good idea! Weirdly though linspace seems to default to floats while arange defaults to (imo nicer) ints:

```python
print np.arange(2, 18, 2).reshape(2, 4)
print np.linspace(2, 16, 8).reshape(2, 4)

[[ 2  4  6  8]
 [10 12 14 16]]
[[  2.   4.   6.   8.]
 [ 10.  12.  14.  16.]]
```

File: course_content/notebooks/numpy_intro.ipynb (hunk 2 above)
Comment (dkillick) [inline course_content/notebooks/numpy_intro.ipynb]: Agreed, I'll get rid of them.

File: course_content/notebooks/numpy_intro.ipynb (hunk 1 above)
Comment (pelson) [inline course_content/notebooks/numpy_intro.ipynb]: I can live with that...

**Comment by dkillick** [comment]
New commit: clarifies where exercise 2 starts following some confusion on that matter...

**Pull request merged**